app/models/user.rb

```diff
@@ -51,4 +56,23 @@ def toggle_admin!
     team = Registry.first.global_namespace.team
     admin ? team.owners << self : team.owners.delete(self)
   end
+
+  ##
+  # Disabling users.
+
+  # Disable this user and remove all its associations with teams.
+  def disable!
+    return unless update_attributes(enabled: false)
+    TeamUser.where(user: id).delete_all
```

Comment:
IMHO this is unexpected. If I delete a user I expect him to be removed from all the teams he was involved with. 

Disabling a user is something which can be reverted. As a user I would expect everything to be unchanged as soon as I am enabled again.

To make a parallel: when you delete your account from facebook all your connections and data is deleted (not sure about the last one :D). When you deactivate your account everything is simply frozen, when you re-enable your account your connections and data are still in place.

We should do something like that.